### PR TITLE
chore: wrap error objects to include context (#10592)

### PR DESCRIPTION
**Pull request by nirsht**
Signed-off-by: Nir Shtein <89006520+nirsht@users.noreply.github.com>

### CODE REVIEW

**Comment by nirsht** [comment]
This PR is related to this issue https://github.com/argoproj/argo-cd/issues/10592

File: cmd/argocd/commands/admin/cluster.go [hunk 1]
```diff
@@ -415,9 +413,8 @@ func NewClusterDisableNamespacedMode() *cobra.Command {
 					cluster.Namespaces = nil
 					fmt.Printf("Disabling namespaced mode for cluster %s...", server)
 					if !dryRun {
-						_, err = argoDB.UpdateCluster(ctx, cluster)
-						if err != nil {
-							return err
+						if _, err = argoDB.UpdateCluster(ctx, cluster); err != nil {
+							return fmt.Errorf("error updating cluster, cluster name: %s, err: %w", cluster.Name, err)
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/admin/cluster.go:417]:
```suggestion
							return fmt.Errorf("error updating cluste: %w", err)
```

File: cmd/argocd/commands/app.go [hunk 2]
```diff
@@ -2427,12 +2427,12 @@ func NewApplicationEditCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
...
 				input, err = yaml.YAMLToJSON(input)
 				if err != nil {
-					return err
+					return fmt.Errorf("error converting YAML to JSON: %w", err)
 				}
 				updatedSpec := argoappv1.ApplicationSpec{}
 				err = json.Unmarshal(input, &updatedSpec)
 				if err != nil {
-					return err
+					return fmt.Errorf("error unmarshal input into updatedSpec: %w", err)
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/app.go:2435]:
```suggestion
					return fmt.Errorf("error unmarshaling input into application spec: %w", err)
```

File: cmd/argocd/commands/project.go [hunk 3]
```diff
@@ -863,23 +863,23 @@ func NewProjectEditCommand(clientOpts *argocdclient.ClientOptions) *cobra.Comman
...
 				input, err = yaml.YAMLToJSON(input)
 				if err != nil {
-					return err
+					return fmt.Errorf("error converting YAML to JSON: %w", err)
 				}
 				updatedSpec := v1alpha1.AppProjectSpec{}
 				err = json.Unmarshal(input, &updatedSpec)
 				if err != nil {
-					return err
+					return fmt.Errorf("error unmarshal input into updatedSpec: %w", err)
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/project.go:871]:
```suggestion
					return fmt.Errorf("error unmarshaling input into application spec: %w", err)
```

File: cmd/argocd/commands/project.go [hunk 4]
```diff
@@ -863,23 +863,23 @@ func NewProjectEditCommand(clientOpts *argocdclient.ClientOptions) *cobra.Comman
...
 				}
 				proj, err := projIf.Get(ctx, &projectpkg.ProjectQuery{Name: projName})
 				if err != nil {
-					return err
+					return fmt.Errorf("could not get project by project name, projName: %s, err: %w", projName, err)
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/project.go:875]:
```suggestion
					return fmt.Errorf("could not get project by project name: %w", err)
```

The project name was user input, so no need to echo it back.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Wording nitpicks.  :-)

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks @nirsht!

**Pull request merged**